tests/test_validators.py

```diff
@@ -637,3 +637,16 @@ def test_validator(self, v):
     with pytest.raises(ConfigError) as exc_info:
         make_generic_validator(test_validator)
     assert ': (self, v), "self" not permitted as first argument, should be: (cls, value' in str(exc_info.value)
+
+
+def test_assert_raises_validation_error():
+    class Model(BaseModel):
+        a: str
+
+        @validator('a')
+        def check_a(cls, v):
+            assert False, 'invalid a'
+
+    with pytest.raises(ValidationError) as exc_info:
+        Model(a='snap')
+    assert 'invalid a' in str(exc_info.value)
```

Comment:
@abdusco 

Given this is the only place where the `AssertionError`->`ValidationError` path is explicitly checked (I think?), it's probably worth adding a check that the entire errors payload is right:

```suggestion
    assert exc_info.value.errors() == [{'loc': ('a',), 'msg': 'invalid a', 'type': 'assertion_error'}]
```

I think right now the type will actually become `'assertionerror'`; it's probably worth updating this function as well (to explicitly handle AssertionError):
https://github.com/samuelcolvin/pydantic/blob/18d4b2bb2ac9790bd58780dba67947a088725731/pydantic/error_wrappers.py#L107

Reply:
Should we count AssertionError as a `value_error` https://github.com/samuelcolvin/pydantic/blob/18d4b2bb2ac9790bd58780dba67947a088725731/pydantic/error_wrappers.py#L109 or should I expand the definition to return an `assertion_error` as well?

Reply:
I figured expand to return 'assertion_error', not just count AssertionError as 'value_error', but @samuelcolvin may have an opinion.

Reply:
I agree `assertion_error`.

Reply:
I'm getting this strange behavior when I try to test using `exc_info.value.errors`:

```
test_validators.py::test_assert_raises_validation_error FAILED           [100%]
tests\test_validators.py:641 (test_assert_raises_validation_error)
[{'loc': ('a',), 'msg': 'invalid a
assert False', 'type': 'assertion_error'}] != [{'loc': ('a',), 'msg': 'invalid a', 'type': 'assertion_error'}]
```

Note that `'msg'` attribute of the exception consist of two lines, and the second one is the assert statement. I'm not sure where it's added. Because when I manually print an AssertionError like this:
```py
try:
    assert False, 'invalid a'
except AssertionError as e:
    print(str(e))
```
it gives `'invalid a'` as expected, but the test suite gives me 
```
invalid a
assert False
```

Reply:
Ok, it comes from here: https://github.com/samuelcolvin/pydantic/blob/d5d9655fc7ed51b95a8123b77c89609cd9156575/pydantic/fields.py#L445

I think `pytest` mutates AssertionErrors for its own internals, because running this and checking the exception in `_apply_validators` method manually I get `str(exc)` as `invalid a` as expected:

```py
from pydantic import BaseModel, validator

class AModel(BaseModel):
    a: str

    @validator('a')
    def validate_a(cls, v):
        assert False, 'invalid a'

if __name__ == '__main__':
    try:
        AModel(a='')
    except Exception as e:
        print(str(e))
```

Reply:
I'm going to write the test this bug? as `'msg': 'invalid a\nassert False'` and investigate further why `pytest` does that. I'll open an issue if need be.